src/update-feeds-loops/update-feeds-loops.ts

```diff
@@ -262,24 +271,26 @@ export const processBatch = async (
     blockNumber,
   });
   const {
-    config: { sponsorWalletMnemonic, chains, deviationThresholdCoefficient },
+    config: { sponsorWalletMnemonic, chains, deviationThresholdCoefficient, signedApiUrls: configSignedApiBaseUrls },
   } = getState();
   const { contracts } = chains[chainId]!;
 
-  updateState((draft) => {
-    for (const dataFeed of batch) {
-      const receivedUrls = zip(dataFeed.signedApiUrls, dataFeed.beaconsWithData).map(([url, beacon]) => ({
-        url: `${url}/${beacon!.airnodeAddress}`,
-        airnodeAddress: beacon!.airnodeAddress,
-      }));
-      if (!draft.signedApiUrls) draft.signedApiUrls = {};
-      if (!draft.signedApiUrls[chainId]) draft.signedApiUrls[chainId] = {};
-      if (!draft.signedApiUrls[chainId]![providerName]) draft.signedApiUrls[chainId]![providerName] = {};
-      for (const { airnodeAddress, url } of receivedUrls) {
-        draft.signedApiUrls[chainId]![providerName]![airnodeAddress] = url;
-      }
-    }
-  });
+  // Generate signed API URLs for the batch
+  const signedApiUrls = batch
+    .map((dataFeed) =>
+      dataFeed.beaconsWithData.map((beacon, index) => {
+        const configSignedApiUrls = configSignedApiBaseUrls.map((baseUrl) => `${baseUrl}/${beacon.airnodeAddress}`);
```

Comment:
I preferred to generate all airnode-populated signedApiUrls here. So base signedApiUrl coming from config also utilized here. With doing this, I simplied the `data-fetcher-loop`. Now, its only responsibility is getting airnode-populated signed API URLs from the state and making the requests to them.

Reply:
I realised this has a small drawback that whenever processing the batch fails (e.g. temporary network error) we won't use the signed API URLs from the config. The previous approach had a different drawback - the Signed APIs for no longer active dAPIs were never cleaned up. 

I think the new behaviour is also OK, but I think it would deserve a comment.